android/src/main/java/com/github/wuxudong/rncharts/listener/RNOnChartValueSelectedListener.java

```diff
@@ -28,13 +30,24 @@ public RNOnChartValueSelectedListener(Chart chart) {
     public void onValueSelected(Entry entry, Highlight h) {
 
         if (mWeakChart != null) {
-            Chart chart = mWeakChart.get();
+            final Chart chart = mWeakChart.get();
 
             ReactContext reactContext = (ReactContext) chart.getContext();
             reactContext.getJSModule(RCTEventEmitter.class).receiveEvent(
                     chart.getId(),
                     "topSelect",
                     EntryToWritableMapUtils.convertEntryToWritableMap(entry));
+            
+            // Hack: unhighlight all values to trigger callback on re-taps
```

Comment:
wouldn't it trigger `onNothingSelected` on de-select? I could be misunderstanding it.

Reply:
right, the desired behavior for our UX though is that if you click on retirement icon, then close the edit-tray, and click on the retirement icon again, we want to still trigger the `onSelect` callback. (vs. in original code, the second click on retirement icon will trigger `onNothingSelected` and will have no indication on which icon was deselected)

Reply:
i see. why not call `chart.highlightValue(null, false);` when we close the edit-tray?

Reply:
makes sense! will update! 🙇‍♀️

Reply:
Thanks for the suggestions @idealllee!
Exposed `resetHighlightValue` instead, and now just calling `setNativeProps` to update this whenever we get the `onSelect` event on a valid data point (see commit https://github.com/indiemoney/snowstorm/pull/776/commits/da6787d897abe74b9a27abf22dd025e298e27dc0)